src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,113 +27,133 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
         super(methodBody, definition);
     }
 
+    @Override
+    protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<DataFlowAbstraction> out) {
+        detectConflict(in, u);
+        FlowSet<DataFlowAbstraction> temp = new ArraySparseSet<>();
+
+        FlowSet<DataFlowAbstraction> killSet = new ArraySparseSet<>();
+
+        for (DataFlowAbstraction item : in) {
+            if (statementEquals(item, u)) {
+                killSet.add(item);
+            }
+        }
+        in.difference(killSet, temp);
+        temp.union(gen(u, in), out);
+    }
+
     @Override
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
         if (isSourceStatement(u) || isSinkStatement(u)) {
-            u.getUseAndDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-                Statement stmt = isSourceStatement(u) ? findSourceStatement(u) : findSinkStatement(u);
-                res.add(new DataFlowAbstraction((Local) v.getValue(), stmt));
-            });
-        } else if (u.getDefBoxes().size() > 0) {
-
-            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
-                String localName = getLocalName((Local) v.getValue());
-
-                for (DataFlowAbstraction defsIn: in){
-                    String defInName = getLocalName(defsIn.getLocal());
-                    //if u not in IN, then add it
-                    if (!defInName.equals(localName)){
-                        res.add(new DataFlowAbstraction(defsIn.getLocal(), defsIn.getStmt())); //update an element in IN
-                        break; //Do not necessary check others elements
-                    }
-                }
+            u.getDefBoxes().stream().filter(v -> v.getValue() instanceof Local || v.getValue() instanceof JArrayRef).forEach(v -> {
+                Statement stmt = isSourceStatement(u)
```

Comment:
No effort, can we create a method isRightStatement that calls isSourceStatement?

It was just the use of a method that already existed.

Reply:
I preferred to create auxiliary methods that internally continue to call `isSourceStatement ` or `isSinkStatement ` because directly changing the method in reaching definition would affect other existing analysis.

Even though I know that the analysis will be changed to not use `ReachDefinition` anymore, I preferred not to send a PR with failed tests. When the analysis are migrated, I think we can reassess that solution.